packages/web/src/components/AddPlayerForm.tsx

```diff
@@ -0,0 +1,151 @@
+import { useState } from 'react';
+
+type ParticipantFormData = {
+  name: string;
+  videoUrl: string;
+}
+
+type FormErrors = {
+  name?: string;
+  videoUrl?: string;
+}
+
+const ParticipantForm = () => {
+  const [formData, setFormData] = useState<ParticipantFormData>({
+    name: '',
+    videoUrl: ''
+  });
+
+  const [players, setPlayers] = useState<ParticipantFormData[]>([]);
+  const [errors, setErrors] = useState<FormErrors>({});
+
+  const isValidYouTubeUrl = (url: string): boolean => {
+    try {
+      const urlObj = new URL(url);
+      
+      // Check for standard youtube.com URLs
+      if (urlObj.hostname === 'www.youtube.com' || urlObj.hostname === 'youtube.com') {
+        const videoId = urlObj.searchParams.get('v');
+        return !!videoId;
+      }
+      
+      // Check for shortened youtu.be URLs
+      if (urlObj.hostname === 'youtu.be') {
+        const videoId = urlObj.pathname.slice(1); // Remove the leading slash
+        return !!videoId;
+      }
+
+      return false;
+    } catch {
+      return false;
+    }
+  };
+
+  // Extract video ID from URL (useful for future API integration)
+  const getYouTubeVideoId = (url: string): string | null => {
+    try {
+      const urlObj = new URL(url);
+      
+      if (urlObj.hostname === 'www.youtube.com' || urlObj.hostname === 'youtube.com') {
+        return urlObj.searchParams.get('v');
+      }
+      
+      if (urlObj.hostname === 'youtu.be') {
+        return urlObj.pathname.slice(1);
+      }
+      
+      return null;
+    } catch {
+      return null;
+    }
+  };
+
+  const validateForm = (): boolean => {
+    const newErrors: FormErrors = {};
+    
+    if (!formData.name.trim()) {
+      newErrors.name = 'Name is required';
+    }
+
+    if (!formData.videoUrl.trim()) {
+      newErrors.videoUrl = 'Video URL is required';
+    } else if (!isValidYouTubeUrl(formData.videoUrl)) {
+      newErrors.videoUrl = 'Please enter a valid YouTube URL';
+    }
+
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData(prev => ({
+      ...prev,
+      [name]: value
+    }));
+    
+    if (errors[name as keyof FormErrors]) {
+      setErrors(prev => ({
+        ...prev,
+        [name]: undefined
+      }));
+    }
+  };
+
+  const addPlayer = (player: ParticipantFormData) => {
+    // You could store the video ID here for future API use
+    const videoId = getYouTubeVideoId(player.videoUrl);
```

Comment:
this seems to be unused

Reply:
Similar to the styles I set myself up to integrate the YouTube API but ended up deciding to wait. Thanks for pointing out that this can be cleaned up.